ci/logs.py

```diff
@@ -94,6 +82,79 @@ def get_and_zip_logs(
                     steps=task_steps,
                 )
             )
+    return task_run_infos
+
+
+def get_failed_excerpts(
+    namespace: str,
+    pipeline_run: dict[str, any],
+    pipeline_run_name: str,
+    only_failed: bool,
+    repo_dir: str,
+    lines: int,
+):
+    task_run_infos = _get_task_run_infos(
+        pipeline_run=pipeline_run,
+        pipeline_run_name=pipeline_run_name,
+        only_failed=only_failed,
+    )
+
+    k8s = client.CoreV1Api()
+    file_path = os.path.join(repo_dir, 'failed_summary.txt')
+    with open(file_path, "w") as out_file:
+        for run_info in task_run_infos:
+            pod_name = run_info.pod_name
+            for idx, (step, container_name) in enumerate(run_info.steps.items()):
+                logger.info(
+                    f'Getting logs for pod {pod_name}, step {step} in container {container_name}'
+                )
+                out_file.write(f'### Pod: {pod_name}, step: {step}, container: {container_name}')
+                try:
+                    data, status_code, headers = k8s.read_namespaced_pod_log_with_http_info(
+                        name=pod_name,
+                        container=container_name,
+                        namespace=namespace,
+                        tail_lines=lines,
+                    )
+
+                    if status_code >= 400:
+                        logger.warning(f'Getting logs failed with {status_code=}')
+                        continue
+
+                    out_file.write(data)
+                except ApiException as ex:
+                    if ex.status == 404:
+                        msg = f'Log for pod {pod_name} is already gone {str(ex)}'
```

Comment:
Converting to string is not necessary here, it will be done by default.

Not necessary to change, just an fyi